confmap/confmap.go

```diff
@@ -209,3 +209,10 @@ func mapKeyStringToMapKeyTextUnmarshalerHookFunc() mapstructure.DecodeHookFuncTy
 		return data, nil
 	}
 }
+
+// Unmarshaler interface may be implemented by types to customize their behavior when being unmarshaled from a Conf.
+type Unmarshaler interface {
+	// Unmarshal is a function that unmarshals a confmap.Conf into the struct in a custom way.
+	// The confmap.Conf for this specific component may be nil or empty if no config available.
```

Comment:
`s/confmap.Conf/Conf` and some minor rewording
```suggestion
	// Unmarshal a Conf into the struct in a custom way.
	// The Conf for this specific component may be nil or empty if no config available.
```